run/events_pubsub/main.go

```diff
@@ -21,46 +21,26 @@ import (
 	"context"
 	"fmt"
 	"log"
-	"net/http"
-	"os"
 
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 )
 
-var (
-	handler = http.DefaultServeMux
-)
-
 func main() {
 	ctx := context.Background()
-	// Create a new HTTP client for CloudEvents
 	p, err := cloudevents.NewHTTP()
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to create protocol: %s", err.Error())
 	}
-	handleFn, err := cloudevents.NewHTTPReceiveHandler(ctx, p, HelloPubSub)
+
+	c, err := cloudevents.NewClient(p)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to create client, %v", err)
 	}
-	handler.Handle("/", handleFn)
 
-	// Determine port for HTTP service.
-	port := os.Getenv("PORT")
-	if port == "" {
-		port = "8080"
-		log.Printf("Defaulting to port %s", port)
-	}
-	// Start HTTP server.
-	log.Printf("Listening on port %s", port)
-	if err := http.ListenAndServe(":"+port, nil); err != nil {
-		log.Fatal(err)
-	}
+	log.Printf("Listening on http://localhost:8080\n")
+	log.Fatalf("Failed to start receiver: %s", c.StartReceiver(ctx, HelloPubSub))
```

Comment:
I'm concern with the approach of using the CloudEvents SDK to manage the HTTP server. While it makes sense as a sort of CloudEvent version of a Functions Framework, in the context of Cloud Run it prevents us from showing developers how to create a fully-functional web server that happens to have some routes that handle CloudEvents.

This makes sense in a quickstart where the goal is more about getting something running, but if this is code we are going to use to show developers how to build CloudEvents support into a Cloud Run service, I think we need to present more flexibility.